app/api/entities/managerFunctions.ts

```diff
@@ -0,0 +1,219 @@
+import { groupBy } from 'lodash';
+import { WithId } from 'api/odm';
+import { attachmentsPath, uploadsPath, files as filesAPI, storeFile } from 'api/files';
+import { processDocument } from 'api/files/processDocument';
+import { search } from 'api/search';
+import { errorLog } from 'api/log';
+import { prettifyError } from 'api/utils/handleError';
+import { FileType } from 'shared/types/fileType';
+import { MetadataObjectSchema } from 'shared/types/commonTypes';
+import { EntityWithFilesSchema } from 'shared/types/entityType';
+import { ClientEntitySchema } from 'app/istore';
+
+const prepareNewFiles = async (
+  entity: EntityWithFilesSchema,
+  updatedEntity: EntityWithFilesSchema,
+  newAttachments: FileType[] = [],
+  newDocuments: FileType[] = []
+) => {
+  const attachments: FileType[] = [];
+  const documents: FileType[] = [];
+
+  const newUrls = entity.attachments?.filter(attachment => !attachment._id && attachment.url);
+
+  if (newAttachments.length) {
+    await Promise.all(
+      newAttachments.map(async (file: FileType) => {
+        const savedFile = await storeFile(attachmentsPath, file, true);
+        attachments.push({
+          ...savedFile,
+          entity: updatedEntity.sharedId,
+          type: 'attachment',
```

Comment:
In general, for this file: would it make sense to lift out the file type enum in `fileSchema.ts` to be importable separately, so we can use that in cases like this, instead of the hard-coded strings? If you agree, but you think this pr is big enough as it is (I think so), we can also add a tech debt issue for that.

Reply:
It makes total sense 👍 , I'll do it here since it's not a big change